lib/url.js

```diff
@@ -213,6 +213,13 @@ export function parseNwcUrl (walletConnectUrl) {
   return params
 }
 
+export function assertContentTypeJson (res) {
+  const contentType = res.headers.get('content-type')
+  if (!contentType || !contentType.includes('application/json')) {
+    throw new Error(`POST ${res.url}: ${res.status} ${res.statusText}`)
```

Comment:
no more gigantic error messages about how we couldn't parse the HTML body as JSON when a reverse proxy throws a gateway timeout